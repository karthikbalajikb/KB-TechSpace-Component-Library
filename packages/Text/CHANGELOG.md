# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.1.0 (2019-09-11)

### Features

- **Text:** add text component ([838960f](https://github.com/karthikbalajikb/KB-TechSpace-Component-Library/commit/838960f))
