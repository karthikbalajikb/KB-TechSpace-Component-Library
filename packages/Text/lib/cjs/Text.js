'use strict';
var __makeTemplateObject =
  (this && this.__makeTemplateObject) ||
  function(cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', { value: raw });
    } else {
      cooked.raw = raw;
    }
    return cooked;
  };
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, '__esModule', { value: true });
var react_1 = __importDefault(require('react'));
var prop_types_1 = __importDefault(require('prop-types'));
var styled_components_1 = __importDefault(require('styled-components'));
var Text = function(_a) {
  var fontSize = _a.fontSize,
    lineHeight = _a.lineHeight,
    letterSpacing = _a.letterSpacing,
    fontWeight = _a.fontWeight,
    color = _a.color,
    opacity = _a.opacity,
    children = _a.children;
  return react_1.default.createElement(
    StyledText,
    {
      fontSize: fontSize || 14,
      lineHeight: lineHeight || 27,
      letterSpacing: letterSpacing,
      fontWeight: fontWeight,
      color: color,
      opacity: opacity,
    },
    children,
  );
};
Text.propTypes = {
  fontSize: prop_types_1.default.number,
  lineHeight: prop_types_1.default.number,
  letterSpacing: prop_types_1.default.number,
  fontWeight: prop_types_1.default.number,
  color: prop_types_1.default.string,
  opacity: prop_types_1.default.number,
  children: prop_types_1.default.any.isRequired,
};
Text.defaultProps = {
  fontSize: 0,
  lineHeight: 0,
  fontWeight: 400,
  letterSpacing: 0.3,
  color: 'lightgray',
  opacity: 1,
};
exports.default = Text;
var StyledText = styled_components_1.default.span(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      [
        '\n  font-size: ',
        ';\n  line-height: ',
        ';\n  letter-spacing: ',
        ';\n  color: ',
        ';\n  opacity: ',
        ';\n',
      ],
      [
        '\n  font-size: ',
        ';\n  line-height: ',
        ';\n  letter-spacing: ',
        ';\n  color: ',
        ';\n  opacity: ',
        ';\n',
      ],
    )),
  function(_a) {
    var fontSize = _a.fontSize;
    return fontSize + 'px';
  },
  function(_a) {
    var lineHeight = _a.lineHeight;
    return lineHeight + 'px';
  },
  function(_a) {
    var letterSpacing = _a.letterSpacing;
    return letterSpacing + 'px';
  },
  function(_a) {
    var color = _a.color;
    return '' + color;
  },
  function(_a) {
    var opacity = _a.opacity;
    return '' + opacity;
  },
);
var templateObject_1;
//# sourceMappingURL=Text.js.map
