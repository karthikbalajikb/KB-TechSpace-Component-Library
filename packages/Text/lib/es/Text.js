var __makeTemplateObject =
  (this && this.__makeTemplateObject) ||
  function(cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', { value: raw });
    } else {
      cooked.raw = raw;
    }
    return cooked;
  };
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
var Text = function(_a) {
  var fontSize = _a.fontSize,
    lineHeight = _a.lineHeight,
    letterSpacing = _a.letterSpacing,
    fontWeight = _a.fontWeight,
    color = _a.color,
    opacity = _a.opacity,
    children = _a.children;
  return React.createElement(
    StyledText,
    {
      fontSize: fontSize || 14,
      lineHeight: lineHeight || 27,
      letterSpacing: letterSpacing,
      fontWeight: fontWeight,
      color: color,
      opacity: opacity,
    },
    children,
  );
};
Text.propTypes = {
  fontSize: PropTypes.number,
  lineHeight: PropTypes.number,
  letterSpacing: PropTypes.number,
  fontWeight: PropTypes.number,
  color: PropTypes.string,
  opacity: PropTypes.number,
  children: PropTypes.any.isRequired,
};
Text.defaultProps = {
  fontSize: 0,
  lineHeight: 0,
  fontWeight: 400,
  letterSpacing: 0.3,
  color: 'lightgray',
  opacity: 1,
};
export default Text;
var StyledText = styled.span(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      [
        '\n  font-size: ',
        ';\n  line-height: ',
        ';\n  letter-spacing: ',
        ';\n  color: ',
        ';\n  opacity: ',
        ';\n',
      ],
      [
        '\n  font-size: ',
        ';\n  line-height: ',
        ';\n  letter-spacing: ',
        ';\n  color: ',
        ';\n  opacity: ',
        ';\n',
      ],
    )),
  function(_a) {
    var fontSize = _a.fontSize;
    return fontSize + 'px';
  },
  function(_a) {
    var lineHeight = _a.lineHeight;
    return lineHeight + 'px';
  },
  function(_a) {
    var letterSpacing = _a.letterSpacing;
    return letterSpacing + 'px';
  },
  function(_a) {
    var color = _a.color;
    return '' + color;
  },
  function(_a) {
    var opacity = _a.opacity;
    return '' + opacity;
  },
);
var templateObject_1;
//# sourceMappingURL=Text.js.map
