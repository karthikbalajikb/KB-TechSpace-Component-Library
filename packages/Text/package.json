{
  "name": "@kb-techspace-org/text",
  "version": "1.0.0",
  "description": "Text Component",
  "author": "Karthik Balaji <karthik@auzmor.com>",
  "homepage": "https://github.com/karthikbalajikb/KB-TechSpace-Component-Library#readme",
  "license": "ISC",
  "main": "lib/Text.js",
  "publishConfig": {
    "access": "public"
  },
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/karthikbalajikb/KB-TechSpace-Component-Library.git"
  },
  "scripts": {
    "test": "echo \"Error: run tests from root\" && exit 1",
    "start": "tsc -w"
  },
  "bugs": {
    "url": "https://github.com/karthikbalajikb/KB-TechSpace-Component-Library/issues"
  }
}
