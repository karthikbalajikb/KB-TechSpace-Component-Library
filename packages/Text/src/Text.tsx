import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

interface Props {
  fontSize?: number;
  lineHeight?: number;
  letterSpacing?: number;
  fontWeight?: number;
  color?: string;
  opacity?: number;
  children?: any;
}

const Text: React.FC<Props> = ({
  fontSize,
  lineHeight,
  letterSpacing,
  fontWeight,
  color,
  opacity,
  children,
}) => (
  <StyledText
    fontSize={fontSize || 14}
    lineHeight={lineHeight || 27}
    letterSpacing={letterSpacing}
    fontWeight={fontWeight}
    color={color}
    opacity={opacity}
  >
    {children}
  </StyledText>
);

Text.propTypes = {
  fontSize: PropTypes.number,
  lineHeight: PropTypes.number,
  letterSpacing: PropTypes.number,
  fontWeight: PropTypes.number,
  color: PropTypes.string,
  opacity: PropTypes.number,
  children: PropTypes.any.isRequired,
};

Text.defaultProps = {
  fontSize: 0,
  lineHeight: 0,
  fontWeight: 400,
  letterSpacing: 0.3,
  color: 'lightgray',
  opacity: 1,
};

export default Text;

const StyledText = styled.span<Props>`
  font-size: ${({ fontSize }) => `${fontSize}px`};
  line-height: ${({ lineHeight }) => `${lineHeight}px`};
  letter-spacing: ${({ letterSpacing }) => `${letterSpacing}px`};
  color: ${({ color }) => `${color}`};
  opacity: ${({ opacity }) => `${opacity}`};
`;
