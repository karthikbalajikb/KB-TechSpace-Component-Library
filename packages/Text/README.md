# `Text`

> Text component

## Usage

```js
import React, { Component } from 'react';
import Text from '@kb-techspace-org/text';

export default class App extends Component {
  render() {
    return <Text>Hello all</Text>;
  }
}
```
