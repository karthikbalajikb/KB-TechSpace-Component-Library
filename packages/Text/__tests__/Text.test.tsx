import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Text from '../src/Text';

describe('Text', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<Text>demo</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
