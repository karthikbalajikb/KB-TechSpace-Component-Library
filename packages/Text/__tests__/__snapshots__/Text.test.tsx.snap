// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
.c0 {
  font-size: 14px;
  line-height: 27px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  color: lightgray;
  opacity: 1;
}

<span
  className="c0"
  color="lightgray"
  fontSize={14}
  fontWeight={400}
  letterSpacing={0.3}
  opacity={1}
>
  demo
</span>
`;
